src/Raven.Server/ServerWide/Commands/CompareExchangeCommands.cs

```diff
@@ -144,22 +145,27 @@ public CompareExchangeResult Execute(ClusterOperationContext context, Table item
 
         public const long InvalidIndexValue = -1;
         
-        public unsafe bool Validate(ClusterOperationContext context, Table items, long index, out long currentIndex)
+        public bool Validate(ClusterOperationContext context, Table items, long index, out long currentIndex)
         {
-            currentIndex = InvalidIndexValue ;
+            currentIndex = InvalidIndexValue;
+            
+            if (index == InvalidIndexValue)
+                return true;
+            
             using (Slice.From(context.Allocator, ActualKey, out Slice keySlice))
             {
-                if (items.ReadByKey(keySlice, out var reader))
+                BlittableJsonReaderObject value;
+                (currentIndex, value) = ClusterStateMachine.GetCompareExchangeValue(context, keySlice, items);
+                if(currentIndex == InvalidIndexValue)
+                    return index == 0;
+                
+                using (value)
                 {
-                    currentIndex = *(long*)reader.Read((int)ClusterStateMachine.CompareExchangeTable.Index, out var _);
-
-                    if (index == InvalidIndexValue )
+                    if (CompareExchangeExpirationStorage.TryGetExpires(value, out var ticks) && ticks < SystemTime.UtcNow.Ticks)
```

Comment:
you can't use here `SystemTime.UtcNow.Ticks`, it might be a different value on a different machines, so your state machine may contain not consistent state.
You need to pass this value from the command caller. (note that we don't care here about accuracy, but just about having a deterministic result across all nodes)

Reply:
Done.
When implementing that I wondered if I should enable edit compare exchange if expired even if it is not atomic guard.
Since changing the de/serialized content is irreversible due to backward compatibility
and the main cause was to allow creating an atomic guarded document if was deleted
I decided to not apply this on regular compare exchange creation.